source/shared/cpp/ObjectModel/CollectionTypeElement.cpp

```diff
@@ -0,0 +1,147 @@
+#include "pch.h"
+#include "CollectionTypeElement.h"
+#include "Util.h"
+
+using namespace AdaptiveSharedNamespace;
+
+CollectionTypeElement::CollectionTypeElement(CardElementType type, ContainerStyle style, 
+    VerticalContentAlignment alignment) : BaseCardElement(type), m_style(style), 
+    m_verticalContentAlignment(alignment), m_hasPadding(false), m_hasBleed(false), m_parentalId()
+{
+}
+
+ContainerStyle CollectionTypeElement::GetStyle() const
+{
+    return m_style;
+}
+
+void CollectionTypeElement::SetStyle(const ContainerStyle value)
+{
+    m_style = value;
+}
+
+VerticalContentAlignment CollectionTypeElement::GetVerticalContentAlignment() const
+{
+    return m_verticalContentAlignment;
+}
+
+void CollectionTypeElement::SetVerticalContentAlignment(const VerticalContentAlignment value)
+{
+    m_verticalContentAlignment = value;
+}
+
+bool CollectionTypeElement::GetPadding() const
+{
+    return m_hasPadding;
+}
+
+void CollectionTypeElement::SetPadding(const bool value)
+{
+    m_hasPadding = value;
+}
+
+// Applies Padding Flag When appropriate
+void CollectionTypeElement::ConfigPadding(const ParseContext& context)
+{
+    // we set padding when parental style is different from child's, and its style should not be None
+    if(GetStyle() != ContainerStyle::None) 
+    {
+        if(context.GetParentalContainerStyle() != GetStyle())
+        {
+            if(context.GetParentalContainerStyle() == ContainerStyle::None &&
+                GetStyle() == ContainerStyle::Default)
+            {
+                SetPadding(false);
+                return;
+            }
+
+            SetPadding(true);
+            return;
+        }
+    }
+
+    SetPadding(false);
+}
+
+bool CollectionTypeElement::GetBleed(void) const
+{
+    return m_hasBleed;
+}
+
+void CollectionTypeElement::SetBleed(const bool value)
+{
+    m_hasBleed = value;
+}
+
+// Applies Padding Flag When appropriate
+void CollectionTypeElement::ConfigBleed(const AdaptiveCards::ParseContext& context)
+{
+    // we allows bleed when self has padding and at least one parent has padding
+    AdaptiveSharedNamespace::InternalId id = context.GetIDOfParentWithPadding();
+    bool canBleed = GetBleed() && (id != AdaptiveSharedNamespace::InternalId::Invalid);
+    if(canBleed)
```

Comment:
>f(c [](start = 5, length = 3)

inconsistent spacing -- please use `clang-format` :)

Reply:
i'm using it. thanks~ :-)

---
In reply to: [259078124](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259078124) [](ancestors = 259078124)

Reply:
fixed

---
In reply to: [259095449](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259095449) [](ancestors = 259095449,259078124)